### PR TITLE
Text drawer support for open controlled gates

### LINKED ISSUE #3864: add visualization of open controlled gates [text drawer]
In PR #3739, open controlled gates were implemented. With that PR a new keyword argument was added to specify the control state,

`XGate().control(3, ctrl_state=6)`

This creates a 4 qubit gate where the first three qubits are controls and the fourth qubit is the target where the x-gate may be applied. A `ctrl_state` may alternatively be expressed as an integer string,

`XGate().control(3, ctrl_state='110')`

where it is more apparent which qubits are open and closed. In this case when q[0] is open and q[1] and q[2] are closed the x-gate will be applied to q[4].

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #3864

### CODE REVIEW

File: qiskit/visualization/text.py [hunk 1]
```diff
@@ -401,6 +401,23 @@ def __init__(self, top_connect="", bot_connect="", conditional=False):
         self.mid_bck = '─'
 
 
+class OpenBullet(DirectOnQuWire):
+    """ Draws an open bullet (usually with a connector). E.g. the top part of a CX gate.
```
Comment (Cryoris) [inline qiskit/visualization/text.py]:
```suggestion
    """Draws an open bullet (usually with a connector). E.g. the top part of a CX gate.
```

File: qiskit/visualization/text.py [hunk 2]
```diff
@@ -401,6 +401,23 @@ def __init__(self, top_connect="", bot_connect="", conditional=False):
...
+
+    def __init__(self, top_connect="", bot_connect="", conditional=False):
+        super().__init__('o')
+        self.top_connect = top_connect
+        self.bot_connect = '│' if conditional else bot_connect
+        self.mid_bck = '─'
+
+
 class EmptyWire(DrawElement):
     """ This element is just the wire, with no instructions nor operations."""
```
Comment (Cryoris) [inline qiskit/visualization/text.py]:
```suggestion
    """This element is just the wire, with no instructions nor operations."""
```

File: test/python/visualization/test_circuit_text_drawer.py [hunk 3]
```diff
@@ -1868,6 +1868,219 @@ def test_controlled_composite_gate_all(self):
...
+                              "        ┌─┴─┐",
+                              "q_1: |0>┤ H ├",
+                              "        └─┬─┘",
+                              "q_2: |0>──o──",
+                              "          │  ",
+                              "q_3: |0>──■──",
+                              "             "])
+        qr = QuantumRegister(4, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.append(HGate().control(3, ctrl_state='100'), [qr[0], qr[3], qr[2], qr[1]])
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply test/python/visualization/test_circuit_text_drawer.py]: the problem with the dec notation is that is not that obvious how the drawing should look like...

File: test/python/visualization/test_circuit_text_drawer.py [hunk 4]
```diff
@@ -1868,6 +1868,219 @@ def test_controlled_composite_gate_all(self):
...
+        expected = '\n'.join(["        ┌───┐",
+                              "q_0: |0>┤ H ├",
+                              "        └─┬─┘",
+                              "q_1: |0>──o──",
+                              "          │  ",
+                              "q_2: |0>──■──",
+                              "             "])
+        qr = QuantumRegister(3, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.append(HGate().control(2, ctrl_state='10'), [qr[2], qr[1], qr[0]])
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply test/python/visualization/test_circuit_text_drawer.py]: you are right.. I'm still confused in my head about the order.. https://github.com/Qiskit/qiskit-terra/issues/3864#issuecomment-597899121

File: test/python/visualization/test_circuit_text_drawer.py [hunk 5]
```diff
@@ -1868,6 +1868,219 @@ def test_controlled_composite_gate_all(self):
...
+            "q_1: |0>┤ U2(pi,-5pi/8) ├",
+            "        └───────┬───────┘",
+            "q_2: |0>────────o────────",
+            "                │        ",
+            "q_3: |0>────────■────────",
+            "                         "])
+        qr = QuantumRegister(4, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.append(U2Gate(pi, -5 * pi / 8).control(3, ctrl_state='100'),
+                       [qr[0], qr[3], qr[2], qr[1]])
```
[the remark this answers is not in the document]
Comment (1ucian0) [reply test/python/visualization/test_circuit_text_drawer.py]: `ctrl_state='010'` in a942f752e

File: test/python/visualization/test_circuit_text_drawer.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Cryoris) [reply test/python/visualization/test_circuit_text_drawer.py]: I agree, but it was more about the fact that the integer argument gets tested as well, not only the string argument

[the remark this answers is not in the document]
Comment (1ucian0) [reply test/python/visualization/test_circuit_text_drawer.py]: That's a `.control` issue :) there are some tests for checking that.

**Pull request merged**